src/app/components/Page.tsx

```diff
@@ -21,64 +21,51 @@
 import { useAuthContext } from '@/global/utils';
 import { EmotionJSX } from '@emotion/react/types/jsx-namespace';
 import { DnaLoader, css } from '@icgc-argo/uikit';
-import { notFound, redirect } from 'next/navigation';
+import { redirect } from 'next/navigation';
 import useUserRole, { UserRoleList } from '../hooks/useUserRole';
 
-// Nextjs dynamic routing params
-type URLParams = {
-	params: { [key: string]: string | string[] | undefined } | undefined;
-	searchParams: { [key: string]: string | string[] | undefined } | undefined;
-};
+export type PageWithPermissionProps = { permissions: { shortName?: string } };
 
 type AcceptedRole = keyof UserRoleList;
-type PageProps = {
-	Component: Component;
-	acceptedRoles: AcceptedRole[];
-	urlParams: URLParams;
-};
-
-const Page = ({ Component, acceptedRoles, urlParams }: PageProps) => {
-	const { egoJwt } = useAuthContext();
-	const programParam = urlParams?.params?.shortName;
-
-	const programName = typeof programParam === 'string' ? programParam : '';
-
-	const userRoles = useUserRole(egoJwt, programName);
-
-	const isAuthorized = acceptedRoles.some((roleKey) => userRoles[roleKey]);
-
-	if (!egoJwt) {
-		return (
-			<div
-				css={css`
-					display: flex;
-					align-items: center;
-					justify-content: center;
-				`}
-			>
-				<DnaLoader />
-			</div>
-		);
-	} else if (Array.isArray(programName)) {
-		// a url with multiple program names is invalid
-		notFound();
-	} else if (isAuthorized) {
-		return <Component {...urlParams} />;
-	}
-
-	redirect('/403');
-};
 
 // Component type for this project because we use the Emotion compiler
-type Component = (...args: any[]) => EmotionJSX.Element;
+type JSXComponent<T extends object> = (args: T) => EmotionJSX.Element;
```

Comment:
Could we use React.ComponentType here?
It covers react component vs a (more specific I think) jsx component

Reply:
probably, give it a try.